src/NUnitFramework/framework/Constraints/Tolerance.cs

```diff
@@ -21,14 +21,14 @@ public sealed class Tolerance
         private const string NumericToleranceRequired = "A numeric tolerance is required";
 
         /// <summary>
-        /// Returns a default Tolerance object, equivalent to an exact match.
+        /// Returns a default Tolerance object, equivalent to a default matching rules.
         /// </summary>
-        public static Tolerance Default => new Tolerance(0, ToleranceMode.Unset);
+        public static readonly Tolerance Default = new Tolerance(0, ToleranceMode.Unset);
```

Comment:
Not as easily as I would like.  I'll need suggestions on how to equate Two tolerances, as they don't have an Equals() operator now.  Are you happy to have Equals implemented for Tolerance?  I'm not 100% sure how I create equal scenario, I'm assuming I'm testing both the type of the value, and that it thinks it's "Equal" to the value of the Exact tolerance.

I converted to this so that I can reference Equals and be sure Default, or Unset where what's really requested.  Maybe the other way where any submission of 0, Linear is considered exact is correct.

What are your thoughts on direction?

Reply:
There are two objectives here,  "Ensure tolerance is available to EqualityAdapters" and "Is not silently ignored".   The first one is very fine.  For the second we do have some options:  What we have done earlier is 1) Things that are "wrong" or "not best practice", are handled by the NUnit.Analyzers and warned there with suitable refactoring added.   2) In the adapter project I have added feature flags around every possibly breaking change, making it default off in the upcoming version, and when it seems like people have got it, I have switched to default on. Sending flags down to the framework has not been so common though, but could be used. "ToleranceMode:  Strict or Relaxed".   A third option would be not to throw an exception but instead, output a warning/error message.    I think the latter might be best suited here, and in addition, ask @mikkelbu about adding a check to the analyzers.

Reply:
Changing back to a property with just reference compare can be done like:
```
public static Tolerance Default { get; } = new Tolerance(0, ToleranceMode.Unset);
```

Reply:
@OsirisTerje an nunit.analyzer to warn that tolerance will not be used with a codefix to remove the `.Within(...)` is a good idea. The issue already [exists](https://github.com/nunit/nunit.analyzers/issues/344) but it has stalled. I will pick it up myself.

Reply:
That's a very good point @OsirisTerje, and my discussion with @stevenaw about backpatching probably centres around being able to do (1), EqualityAdapters with tolerance support.  That's what's needed for more flexible extension.

My correctness bent is the surprise, it's been like this for a long while without complaint. So only making the tolerance checking changes in master (v4) is a good direction.